crates/ty_python_semantic/src/types/diagnostic.rs

```diff
@@ -353,7 +445,18 @@ declare_lint! {
 }
 
 declare_lint! {
-    /// TODO #14889
+    /// ## What it does
+    /// Checks for class bases which are not a class or of a gradual type
+    /// (`Any`, `Unknown`, `Todo`).
+    ///
+    /// ## Why is this bad?
+    /// Such bases likely indicate a logic error.
+    ///
+    /// ## Examples
+    /// ```python
+    /// class A(1): ...
+    /// class B(A()): ...
+    /// ```
```

Comment:
```suggestion
    /// Checks for classes that have bases which are not themselves
    /// either a class or a [gradual form]
    ///
    /// ## Why is this bad?
    /// Many such class definitions will lead to `TypeError` being raised at runtime.
    ///
    /// ## Examples
    /// ```python
    /// class A(1): ...
    /// class B(A()): ...
    /// ```
    ///
    /// [gradual form]: https://typing.python.org/en/latest/spec/glossary.html#term-gradual-form
```

We should also probably break this diagnostic up into two. The first of these classes will fail at runtime, but the second one won't -- it just can't be properly understood by ty: https://play.ty.dev/5acf6092-7608-4c4f-8e73-989ea03ab029.

We're using the same error code for both class definitions; that seems inappropriate.

Reply:
> Just to be clear, these are all allowed, correct?

oh, great point! I still think "of a gradual type" isn't quite right, since according to [the spec](https://typing.python.org/en/latest/spec/glossary.html#term-gradual-type):

> All types in the Python type system are “gradual”

This conversation is just further convincing me that we should split this lint into two, because there are many types which we currently reject here that are assignable to `type` (and therefore should be valid bases, since they (probably) won't lead to `TypeError` being raised at runtime), but which are disallowed by this lint because we wouldn't be able to infer what the MRO of the resulting class would be. It's two distinct problems that have two distinct consequences, but they're currently covered by the same error code ☹️

Feel free to leave the documentation of this lint for now. I'm not sure it makes sense to add documentation when the lint itself doesn't really feel like it makes sense right now.